src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt

```diff
@@ -640,6 +645,24 @@ class CourtSentencingSynchronisationService(
     log.warn("Unable to delete mapping for court appearance $dpsCourtAppearanceId. Please delete manually", e)
   }
 
+  private suspend fun getDpsChargeMappings(nomisSentence: SentenceResponse): List<String> {
+    return try {
+      nomisSentence.offenderCharges.map {
+        mappingApiService.getOffenderChargeByNomisId(it.id).dpsCourtChargeId
+      }
+    } catch (e: Exception) {
+      telemetryClient.trackEvent(
+        name = "charge-mapping-missing",
```

Comment:
Is this only true for a not found exception? You might get a 503 or something else...

Reply:
true - I'll revisit on the next pr